### PR TITLE
Key value storage trunk

### PR DESCRIPTION
**Pull request by kamilsa**
## What is this pull request?
This PR is composed from several PRs related to implementing key value storage for account's details information.
   
## Why do you implement it? Why do we need this pull request?
Please check if everything related to key-value storage is implemented and ready to be merged to develop
  
## How to use the features provided in the pull request?
Check if everything is working as it should be. Check tests 

## Details/Features

1. New json field containing key-value information in account model
2. Updated sql schema containing key-value field by means of `jsonb` from postgres
3. Updated converters
4. Updated CLI for setting and getting key-value information for account

### CODE REVIEW

File: iroha-cli/interactive/impl/interactive_transaction_cli.cpp [hunk 1]
```diff
@@ -418,6 +422,15 @@ namespace iroha_cli {
           src_account_id, dest_account_id, asset_id, amount);
     }
 
+    std::shared_ptr<iroha::model::Command>
+    InteractiveTransactionCli::parseSetAccountDetail(
+        std::vector<std::string> params) {
+      auto account_id = params[0];
+      auto key = params[1];
+      auto value = params[2];
+      return std::make_shared<SetAccountDetail>(account_id, key, value);
```
Comment (Warchant) [inline iroha-cli/interactive/impl/interactive_transaction_cli.cpp]: what if `params` has no 3 elements? Let's say, 2.
Is it possible?

File: irohad/ametsuchi/impl/postgres_wsv_command.cpp [hunk 2]
```diff
@@ -62,7 +64,8 @@ namespace iroha {
         transaction_.exec(
             "INSERT INTO role_has_permissions(role_id, permission_id) VALUES "
             + std::accumulate(
-                  std::next(permissions.begin()), permissions.end(),
+                  std::next(permissions.begin()),
+                  permissions.end(),
                   entry(*permissions.begin()),
                   [&entry](auto acc, auto x) { return acc + ", " + entry(x); })
```
Comment (Warchant) [inline irohad/ametsuchi/impl/postgres_wsv_command.cpp]: this is not even 1st normal form :(

File: irohad/ametsuchi/impl/postgres_wsv_command.cpp [hunk 3]
```diff
@@ -111,15 +122,19 @@ namespace iroha {
...
+            "INSERT INTO account(account_id, domain_id, quorum, "
+            "transaction_count, data) VALUES ("
+            + transaction_.quote(account.account_id)
+            + ", "
+            + transaction_.quote(account.domain_id)
+            + ", "
+            + transaction_.quote(account.quorum)
+            + ", "
+            // Transaction counter
+            + transaction_.quote(0)
```
Comment (Warchant) [inline irohad/ametsuchi/impl/postgres_wsv_command.cpp]: change to constant which is defined inside `PostgresWsvCommand`

File: test/module/irohad/ametsuchi/kv_storage_test.cpp [hunk 4]
```diff
@@ -0,0 +1,152 @@
...
+#include "model/commands/create_role.hpp"
+#include "model/commands/set_account_detail.hpp"
+#include "model/permissions.hpp"
+#include "module/irohad/ametsuchi/ametsuchi_fixture.hpp"
+
+using namespace iroha::ametsuchi;
+using namespace iroha::model;
+
+/**
+ * Fixture for kv storage test. Creates two account, one containing age
```
Comment (Warchant) [inline test/module/irohad/ametsuchi/kv_storage_test.cpp]: two account**s**

one **has**

File: iroha-cli/interactive/impl/interactive_transaction_cli.cpp (hunk 1 above)
Comment (kamilsa) [reply iroha-cli/interactive/impl/interactive_transaction_cli.cpp]: No, cli would not get here if there were other than 3 elements

File: irohad/ametsuchi/impl/postgres_wsv_command.cpp (hunk 2 above)
Comment (kamilsa) [reply irohad/ametsuchi/impl/postgres_wsv_command.cpp]: it isn't. It inserts multiple rows at once. Not multiple values into single row

**Comment by Warchant** [comment]
Just noticed:
>MAJOR pb_query_response_factory.cpp#L266: Add a default case to this switch. rule

Fix this.

**Pull request merged**